backend/clubs/views.py

```diff
@@ -4841,7 +4883,21 @@ def initiate_checkout(self, request, *args, **kwargs):
         holding_expiration = timezone.now() + datetime.timedelta(minutes=10)
         tickets.update(holder=self.request.user, holding_expiration=holding_expiration)
 
-        cart_total = tickets.aggregate(total=Sum("price"))["total"]
+        # Calculate cart total, applying group discounts where appropriate
+        ticket_type_counts = tickets.values("type").annotate(count=Count("type"))
+        ticket_type_counts = {
+            item["type"]: item["count"] for item in ticket_type_counts
+        }
+
+        cart_total = sum(
+            ticket.price * (1 - ticket.group_discount)
+            if ticket.group_discount
+            and ticket.group_size
+            and ticket_type_counts[ticket.type] >= ticket.group_size
+            else ticket.price
+            for ticket in tickets
+        )
```

Comment:
This is pretty ugly, but I couldn't think of a better way to do it.

Reply:
Setting `ticket.group_discount` to default to 0 is probably a good idea here:
- You can skip the `if ticket.group_discount` check
- `1 - ticket.group_discount` won't fail at runtime if the latter is `None`